oeps/oep-0011.rst

```diff
@@ -0,0 +1,299 @@
+   more details.
+
+2. **How should offline support be provided?**
+
+   For many mobile users, it would be very useful if they could continue
+   their Open edX courses while offline. This is true both for true mobile
+   apps as well as for responsive web applications. This is an area that
+   has not been explored much up to now, but which will only become more
+   important going forward.
+
+3. **Should Universal JavaScript be considered?**
+
+   There are many companies that are seeing performance benefits by
+   pre-rendering their dynamic client-side applications on the server first.
+   The most common approach is to run the same JavaScript code both in the
+   browser and using Node on the server. This requires the use of a JavaScript
+   framework that does not perform direct manipulation of the DOM, but instead
+   works with an abstraction of the desired HTML. For example, Netflix is
+   seeing great performance gains by using `React`_ in this way, as can be
+   seen in the blog post `Making Netflix Faster`_.
+
+4. **Where do web components fit with Open edX applications?**
+
+   Web Components is a W3C recommendation that allows developers to create
+   custom HTML elements for use within their web applications. The
+   `Polymer Project` provides a library that allows web components to be
+   built in most modern browsers. It seems that there is a useful overlap
+   between web components and Open edX's `XBlocks`_
+
+5. **Which build tools and asset pipeline should be used?**
+
+   There are already a variety of technologies used within Open edX,
+   none of which is a clear winner. Django Pipeline is used by
+   edx-platform, Django Compressor is used by some of the Ecommerce
+   applications, and Webpack is prevalent in pure front-end implementations.
+
+6. **Should HTTP/2 be used in preference to asset bundling?**
+
+   The Hypertext Transfer Protocol Version 2 (HTTP/2) enables more efficient
+   network communication between clients and servers. In particular, it
+   allows for multiple requests to be handled with the same TCP connection,
+   which removes most of the benefits of asset bundling. It is not yet
+   clear whether this is ready for full scale adoption, but it would reduce
+   a lot of complexity from application's asset pipelines.
+
+
+
+Rejected Alternatives
+=====================
+
+Note: this section needs fleshing out...
+
+.. _Reasons for rejecting Bootstrap:
+1. **Bootstrap**
+
+ - size
+ - not accessible
+ - too inflexible
+ - might want to consider compatibility to allow mixing Bootstrap components
+   into pattern library pages
+
+.. _Reasons for rejecting CoffeeScript:
+2. **CoffeeScript**
+
+   The Open edX codebase made heavy use of CoffeeScript for several years,
+   but it has now been officially deprecated. Most of the language
+   benefits of CoffeeScript are now made available via ES2015.
+
+.. _Reasons for rejecting Mako:
+3. **Mako**
```

Comment:
Was Jinja2 also rejected, or just not considered yet?  I'm not necessarily advocating it, but it's popular enough that it probably deserves a mention here.